x-pack/plugins/lens/public/types.ts

```diff
@@ -131,18 +121,16 @@ export interface VisualizationProps<T = unknown> {
 }
 
 export interface SuggestionRequest<T = unknown> {
-  // Roles currently being used
-  roles: DimensionRole[];
   // It is up to the Visualization to rank these tables
-  tableColumns: { [datasourceSuggestionId: string]: TableColumns };
+  tableColumns: Record<number, TableColumn[]>;
```

Comment:
Only string keys make sense here, right? It's a JS object...

Reply:
numbers are also valid keys in an object, but I get your point ;) . I changed it to number because this magic "id" is just the index in the array of data table suggestions which the datasource provided. I updated the names and made this an array because it's essentially that - plus suggestionId is really misleading here because it's not clear whether we are talking about a datatable suggestion or a visualization suggestion. Please check out my latest commit. If something is still unclear, I'm happy to change it, this could be more intuitive for sure.

The flow is basically like this (this code is located in `suggestion_helpers`):
```
suggestedDataTables: Array = datasource.getSuggestedDatatables();
suggestedVisualizations: Array = visualizations.flatMap(vis => {
  return vis.getVisualizationSuggestions(suggestedDataTables);
});
suggestedVisualizations.sort();
return suggestedVisualizations;
```

The datasourceSuggestionId comes into play to express which of the multiple suggested data tables the visualization wants to use for any given suggestion. It is just the index of the suggestedDataTables array, not something created by the datasource itself or anything. It makes total sense to refer to the same suggested data table in multiple suggestions. Because of this I renamed the `datasourceSuggestionId` to `tableIndex` (https://github.com/elastic/kibana/pull/36513/files#diff-c9e959476594f76ae1ce129e8beb7e50R133) and renamed the confusing `tableColumns` in a suggestion request to `tables` (https://github.com/elastic/kibana/pull/36513/files#diff-c9e959476594f76ae1ce129e8beb7e50R125)